bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/AbstractListItemImpl.java

```diff
@@ -0,0 +1,38 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2017 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.models.v1;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.StringUtils;
+
+public abstract class AbstractListItemImpl extends AbstractDataLayerProvider {
+    protected String parentId;
+    protected String path;
+
+    public AbstractListItemImpl(String parentId, String path) {
+        this.parentId = parentId;
+        this.path = path;
+    }
+
+    public String getId() {
+        return parentId + "-item-" + StringUtils.substring(DigestUtils.sha1Hex(path), 0, 10);
```

Comment:
we should check on parentId being null and return null in this case

Reply:
IMHO we should put this logic in a utils dedicated to ID generation. We would then only have to refactor such specific code in one place.
sha1Hex needs to be sha256Hex